activestorage/app/models/active_storage/blob.rb

```diff
@@ -154,6 +156,12 @@ def filename
     ActiveStorage::Filename.new(self[:filename])
   end
 
+  # Returns the destination format to be used by processors when a variant
+  # image is created.
+  def destination_format
+    MimeMagic.new(content_type).extensions.first
+  end
```

Comment:
@georgeclaghorn Should we always use the `content_type` to determinate the destination format or only if the extension is unknown?